src/Avalonia.Diagnostics/Diagnostics/DevTools.cs

```diff
@@ -74,8 +87,67 @@ public static IDisposable Open(TopLevel root, DevToolsOptions options)
         private static void DevToolsClosed(object? sender, EventArgs e)
         {
             var window = (MainWindow)sender!;
-            s_open.Remove(window.Root!);
             window.Closed -= DevToolsClosed;
+            if (window.Root is Controls.Application host)
+            {
+                s_open.Remove(host.Instance);
+            }
+            else
+            {
+                s_open.Remove(window.Root!);
+            }                       
+        }
+
+        internal static void Attach(Application application, DevToolsOptions options)
+        {
+            // Skip if call on Design Mode
+            if(Avalonia.Controls.Design.IsDesignMode)
+            {
+                return;
+            }
+            if (s_attachedToApplication == true)
+            {
+                throw new ArgumentException("DevTools already attached to application", nameof(application));
```

Comment:
I wonder, should we allow attaching to both application and window, but with different hotkeys? Or it will introduce unnecessary complexity?

Reply:
I think it is not necessary, if the developer needs to link it to only one window he will use the specific overload.